diff --git a/python/dgl/sparse/__init__.py b/python/dgl/sparse/__init__.py
index 3cc2bc4..714b5e3 100644
--- a/python/dgl/sparse/__init__.py
+++ b/python/dgl/sparse/__init__.py
@@ -1,4 +1,5 @@
 """dgl sparse class."""
+import glob
 import os
 import sys
 
@@ -18,19 +19,22 @@ from .unary_op import *
 
 def load_dgl_sparse():
     """Load DGL C++ sparse library"""
-    version = torch.__version__.split("+", maxsplit=1)[0]
 
     if sys.platform.startswith("linux"):
-        basename = f"libdgl_sparse_pytorch_{version}.so"
+        shared_lib_ext = ".so"
     elif sys.platform.startswith("darwin"):
-        basename = f"libdgl_sparse_pytorch_{version}.dylib"
+        shared_lib_ext = ".dylib"
     elif sys.platform.startswith("win"):
-        basename = f"dgl_sparse_pytorch_{version}.dll"
+        shared_lib_ext = ".dll"
     else:
         raise NotImplementedError("Unsupported system: %s" % sys.platform)
 
     dirname = os.path.dirname(libinfo.find_lib_path()[0])
-    path = os.path.join(dirname, "dgl_sparse", basename)
+    path = os.path.join(dirname, "dgl_sparse", f"*{shared_lib_ext}")
+    path = glob.glob(path)
+    if len(path) > 1:
+        raise ImportError("More than one DGL C++ sparse library found")
+    path = path[0]
     if not os.path.exists(path):
         raise FileNotFoundError(f"Cannot find DGL C++ sparse library at {path}")
 
